Configuration.props

```diff
@@ -118,6 +118,10 @@
     <MingwCommandPrefix32>i686-w64-mingw32</MingwCommandPrefix32>
     <MingwCommandPrefix64>x86_64-w64-mingw32</MingwCommandPrefix64>
   </PropertyGroup>
+  <PropertyGroup>
+    <JavacSourceVersion>1.8</JavacSourceVersion>
+    <JavacTargetVersion>1.8</JavacTargetVersion>
```

Comment:
I think we still have support for JDK 1.6/1.7, if we merge this, do we need to require 1.8?

I think the relevant code is here:

https://github.com/xamarin/xamarin-android/blob/c39f1fe4ee43b72d41d3a99b520aa8fdabf9cfd5/src/Xamarin.Android.Build.Tasks/Tasks/Legacy/ValidateJavaVersion.cs#L48-L68

Reply:
Good question, I don't know the answers.

If we use `javac -target 1.8`, the resulting `.class` file requires Java 1.8.  However, is the code impacted by this PR impacted by that?  For example, `src/java-runtime` is only run on-device, not on the build machine, so requiring 1.8 for `src/java-runtime` shouldn't break any existing customers who have Java 1.6 installed.

`jar2xml.jar`, on the other hand, *may* be executed on customer machines.  (I wish that weren't the case, and I thought I saw a suggestion to obsolete & remove for .net5?). https://github.com/xamarin/jar2xml/commit/4125da6 and https://github.com/xamarin/monodroid/pull/1083 would *require* Java 1.8 in order to execute it.

What's the `javac -target` for `src/r8` and `src/manifestmerger`?  I don't see an explicit Language version there, which means we may *already* be requiring Java 1.8 be installed?  I don't know.

Finally, the `<ValidateJavaVersion/>` code you pointed to mentions `AndroidSdkBuildToolsVersion`; that's the Android SDK `build-tools` version, so this may need to be updated to check for build-tools 30 and require Java 11 there.

Reply:
For `r8` and `manifestmerger`, I bet it's doing this by default:
```
java {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
}
```
But we should probably put this in every `build.gradle` file explicitly.

Reply:
If that "bet" is correct, then we're *already* in a scenario where customers *must* be using JDK 8 or they can't use `r8.jar`, which is required (I think?).  Meaning JDK 6/7 support is already gone.

Reply:
Given that using `manifestmerger.jar` *broke* when we didn't have `java.targetCompatibility` set, the bet *is* correct: JDK 8 is already a minimum requirement for all functionality.